Sources/HyphenableText/HyphenableText.swift

```diff
@@ -13,6 +13,26 @@ public extension String {
   ///
   /// - see: [Wikipedia](https://en.wikipedia.org/wiki/Soft_hyphen).
   static let softHyphen = "\u{00AD}"
+    
+  /// Split the string using space as a separator and, for those substrings
+  /// whose lengths are greater than or equal to minimumWordLength,
+  /// replace them with a softHyphenated version.
+  ///
+  /// - note: This assumes that words are delineated by
+  /// space characters, which may not be correct for all locales where
+  /// CFStringIsHyphenationAvailableForLocale(_:) returns true. Consider
+  /// using CFStringTokenizer to eliminate this assumption.
+  func softHyphenateByWord(minimumWordLength: Int = 0, withLocale locale: Locale = .autoupdatingCurrent) -> Self {
+      var substringArray: [String] = self.split(separator: " ", omittingEmptySubsequences: false).map({ String($0) })
```

Comment:
_Nitpicking here, sorry:_ `self` is not needed, right?

Reply:
Ah, perhaps not. My personal style is to include `self` in some places just to improve clarity and readability. But I'll go ahead and remove it.

Reply:
That's alright, I usually avoid it so that I get errors from the compiler when I'm referencing `self` in escaping closures, so that I have to actively think about retain cycles 😃